ocaml-lsp-server/src/prefix_parser.ml

```diff
@@ -0,0 +1,36 @@
+open Re
+
+(*Regex based parser*)
+
+let name_or_label_regex =
+   {|([~?`]|let%|and%)?([a-zA-Z0-9_']|[a-zA-Z0-9_'] *\. *)+$|}
+
+let name_or_label_regex_rev =
+  Re.compile @@ Re.Posix.re {|^([a-zA-Z0-9_']| *\. *[a-zA-Z0-9_'])+([~?`]|%tel|%dna)?|}
+
+(** matches let%lwt and let* style expressions. See here:https://v2.ocaml.org/manual/bindingops.html *)
+let monadic_bind =
+  Re.compile @@ Re.Posix.re {|(let|and)([$&*+\-/=>@^|.]|(%[a-zA-Z0-9_']*))$|}
+let monadic_bind_rev =
+  Re.compile @@ Re.Posix.re {|^([$&*+\-/=>@^|.]|([a-zA-Z0-9_']*)%)(tel|dna)|}
+
+let infixRegex = Re.compile @@ Re.Posix.re {|[~?:!$&*+\-\/=><@^|%<.#]+$|}
```

Comment:
`infix_regex`

Reply:
Also, the `regex` suffix isn't particularly useful. We already know what it is from the type